pkg/auth/dependency/interaction/oob.go

```diff
@@ -0,0 +1,123 @@
+package interaction
+
+import (
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/authenticator/oob"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/urlprefix"
+	taskspec "github.com/skygeario/skygear-server/pkg/auth/task/spec"
+	"github.com/skygeario/skygear-server/pkg/core/async"
+	"github.com/skygeario/skygear-server/pkg/core/authn"
+	"github.com/skygeario/skygear-server/pkg/core/config"
+	"github.com/skygeario/skygear-server/pkg/core/mail"
+	"github.com/skygeario/skygear-server/pkg/core/sms"
+	"github.com/skygeario/skygear-server/pkg/core/template"
+)
+
+type OOBProviderImpl struct {
```

Comment:
I suppose this provider should belong to `authenticator/oob` package, and can accept `*oob.Authenticator` instead?

Reply:
We should accept AuthenticatorSpec because oob.Authenticator is generated after the code is verified.